internal/internal_workflow_test.go

```diff
@@ -757,6 +758,32 @@ func (s *WorkflowUnitTest) Test_CorruptedSignalOnClosedChannelWorkflow_Receive_S
 	s.EqualValues(0, len(result))
 }
 
+func bufferedChanWorkflowTest(ctx Context) error {
+	bufferSize := 1
+	ch := NewBufferedChannel(ctx, bufferSize)
+	for i := 0; i < bufferSize+1; i++ {
+		ch.Send(ctx, i)
+		Go(ctx, func(ctx Context) {
+			var dummy int
+			ch.Receive(ctx, &dummy)
+		})
+	}
+
+	// Refill the buffer
+	for i := 0; i < bufferSize; i++ {
+		ch.Send(ctx, i)
+	}
+	return nil
+}
+
+func (s *WorkflowUnitTest) Test_BufferedChanWorkflow() {
+	env := s.NewTestWorkflowEnvironment()
+
+	env.ExecuteWorkflow(bufferedChanWorkflowTest)
+	s.True(env.IsWorkflowCompleted())
+	s.NoError(env.GetWorkflowError())
+}
+
```

Comment:
Can we add a test or maybe even in the same test that passes buffer size > 1

Reply:
Once bufferSize is a parameter, I can use a loop to test more cases.